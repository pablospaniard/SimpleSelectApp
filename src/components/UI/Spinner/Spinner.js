import React from 'react'
import styled from 'styled-components'

const StyledSpinner = styled.div`
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
  margin: 200px calc(50% - 32px);

  div {
    position: absolute;
    top: 27px;
    width: 11px;
    height: 11px;
    border-radius: 50%;
    background: #f50057;
    animation-timing-function: cubic-bezier(0, 1, 1, 0);

    &:nth-child(1) {
      left: 6px;
      animation: ellipsis1 0.6s infinite;
    }

    &:nth-child(2) {
      left: 6px;
      animation: ellipsis2 0.6s infinite;
    }

    &:nth-child(3) {
      left: 26px;
      animation: ellipsis2 0.6s infinite;
    }

    &:nth-child(4) {
      left: 45px;
      animation: ellipsis3 0.6s infinite;
    }
  }

  @keyframes ellipsis1 {
    0% {
      transform: scale(0);
    }

    100% {
      transform: scale(1);
    }
  }

  @keyframes ellipsis3 {
    0% {
      transform: scale(1);
    }

    100% {
      transform: scale(0);
    }
  }

  @keyframes ellipsis2 {
    0% {
      transform: translate(0, 0);
    }

    100% {
      transform: translate(19px, 0);
    }
  }
`

const Spinner = () => (
  <StyledSpinner data-testid="spinner">
    <div />
    <div />
    <div />
    <div />
  </StyledSpinner>
)

export default Spinner
