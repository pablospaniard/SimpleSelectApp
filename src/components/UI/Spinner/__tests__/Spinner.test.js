import React from 'react'
import { render, cleanup } from 'react-testing-library'
import Spinner from '../Spinner'

describe('Spinner should render properly', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = render(<Spinner />)
    expect(container.firstChild).toMatchSnapshot('Spinner_snapshot_1')
  })
})
