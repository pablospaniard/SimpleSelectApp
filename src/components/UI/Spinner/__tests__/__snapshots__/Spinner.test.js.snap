// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner_snapshot_1 1`] = `
<div
  class="sc-bdVaJa qNlxh"
  data-testid="spinner"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
